File: C++/ps_baekjoon/12886/main.cpp
#include <bits/stdc++.h>
using namespace std;

constexpr int mxN = 5e2;
bool vst[mxN * 3 + 1][mxN * 3 + 1];
int a, b, c;
queue<pair<int, int>> q;

int main()
{
    cin.tie(0)->sync_with_stdio(0);
    cin >> a >> b >> c;
    int total = a + b + c;
    if (total % 3 != 0)
    {
        cout << 0;
        return 0;
    }
    vst[a][b] = true;
    q.emplace(a, b);
    while (!q.empty())
    {
        auto [a, b] = q.front();
        q.pop();
        int c = total - a - b;
        int temp[]{a, b, c};
        for (int i = 0; i < 3; ++i)
            for (int j = 0; j < 3; ++j)
                if (temp[i] < temp[j])
                {
                    int num1 = temp[i] + temp[i];
                    int num2 = temp[j] - temp[i];
                    if (vst[num1][num2])
                        continue;
                    vst[num1][num2] = true;
                    q.emplace(num1, num2);
                }
    }
    cout << (vst[total / 3][total / 3] ? 1 : 0);
    return 0;
}

File: Python/ps_baekjoon/12886/main.py
import sys
from queue import Queue

readline, write = (sys.stdin.readline, sys.stdout.write)
mxN = 500
vst = [[False] * (mxN * 3 + 1) for _ in range(mxN * 3 + 1)]
q = Queue()


a, b, c = [int(i) for i in readline().split()]
total = a + b + c
if total % 3 != 0:
    write("0")
    exit(0)
q.put((a, b))
vst[a][b] = True
while not q.empty():
    a, b = q.get()
    c = total - a - b
    temp = (a, b, c)
    for i in range(3):
        for j in range(3):
            if temp[i] < temp[j]:
                num1 = temp[i] + temp[i]
                num2 = temp[j] - temp[i]
                if vst[num1][num2]:
                    continue
                vst[num1][num2] = True
                q.put((num1, num2))
write(f"{1 if vst[total//3][total//3] else 0}")